code/modules/fish/fishtank.dm

```diff
@@ -346,14 +348,15 @@
 		else if(istype(user.l_hand, /obj/item/storage/bag/fish))
 			fish_bag = user.l_hand
 		var/fish_name = fish_to_scoop.fish_name
-		if(fish_bag)
-			user.visible_message("[user.name] scoops \a [fish_name] from [src] into [fish_bag].", "You scoop \a [fish_name] out of [src] into [fish_bag].")
-		else
-			user.visible_message("[user.name] scoops \a [fish_name] from [src].", "You scoop \a [fish_name] out of [src].")
 		// Move the fish in
 		var/fish_item = fish_to_scoop.fish_item
 		if(fish_item)
-			new fish_item(fish_bag || get_turf(user))	//Spawn the appropriate fish_item at the user's feet or in their held fish bag.
+			var/obj/item/I = new fish_item(get_turf(user))
+			if(fish_bag?.can_be_inserted(I))
+				fish_bag.handle_item_insertion(I)
+			else
+				fish_bag = null
```

Comment:
Do you need to set this to null here? The variable isn't used again after this point.

Reply:
Whoops